source/math/FloatComplexVectorList.ooc

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (C) 2014 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+import FloatComplex
+use ooc-collections
+import FloatVectorList
+
+FloatComplexVectorList: class extends VectorList<FloatComplex> {
+	init: func ~default {
+		super()
+	}
+	init: func ~capacity (capacity: Int) {
+		super()
+		for (i in 0..capacity)
+			this add(FloatComplex new())
+	}
+	init: func ~fromVectorList (other: VectorList<FloatComplex>) {
+		super(other _vector)
+		this _count = other count
+	}
+	toVectorList: func() -> VectorList<FloatComplex> {
+		result := VectorList<FloatComplex> new()
+		result _vector = this _vector
+		result _count = this _count
+		result
+	}
+	sum: func -> FloatComplex {
+		result := FloatComplex new()
+		for (i in 0..this _count)
+			result = result + this[i]
+		result
+	}
+	mean: func -> FloatComplex {
+		sum() / this _count
+	}
+	getReal: func -> FloatVectorList {
```

Comment:
![](http://www.raidz3ro.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/g/e/get-real-be-rational-t-shirt-thumbnail.jpg)

No but seriously, shouldn't this be a property?